dockstore-webservice/src/main/java/io/dockstore/webservice/resources/TokenResource.java

```diff
@@ -562,10 +562,21 @@ private Userinfoplus getUserInfo(String accessToken) {
     public Token addToken(@ApiParam("code") String satellizerJson) {
         Gson gson = new Gson();
         JsonElement element = gson.fromJson(satellizerJson, JsonElement.class);
-        JsonObject satellizerObject = element.getAsJsonObject();
-        final String code = getCodeFromSatellizerObject(satellizerObject);
-        final boolean registerUser = getRegisterFromSatellizerObject(satellizerObject);
-        return handleGitHubUser(null, code, registerUser);
+        System.out.println(element);
+        if (element != null) {
+            try {
+                JsonObject satellizerObject = element.getAsJsonObject();
+                final String code = getCodeFromSatellizerObject(satellizerObject);
+                final boolean registerUser = getRegisterFromSatellizerObject(satellizerObject);
+                return handleGitHubUser(null, code, registerUser);
+            } catch (IllegalStateException ex) {
+                throw new CustomWebApplicationException("Request body is an invalid JSON", HttpStatus.SC_BAD_REQUEST);
```

Comment:
Optionally log the exception, although I'm torn on this one -- if it's not valid JSON, it's probably fishy, and maybe it shouldn't be in our logs. OTOH, maybe it should be.

Reply:
I'd probably say no/not worth it, at least not without thorough escaping/sanitation and this should eventually caught at a higher level, like the firewall or nginx or something.